test/unit/storage/storage.spec.ts

```diff
@@ -29,88 +29,110 @@ describe('Storage', () => {
-      }).to.throw(expectedError);
-    });
-
-    it('should not throw given a valid app', () => {
-      expect(() => {
-        return new Storage(mockApp);
-      }).not.to.throw();
+  
+    describe('app', () => {
+      it('returns the app from the constructor', () => {
+        // We expect referential equality here
+        expect(storage.app).to.equal(mockApp);
+      });
+  
+      it('is read-only', () => {
+        expect(() => {
+          (storage as any).app = mockApp;
+        }).to.throw('Cannot set property app of #<Storage> which has only a getter');
+      });
     });
-  });
-
-  describe('app', () => {
-    it('returns the app from the constructor', () => {
-      // We expect referential equality here
-      expect(storage.app).to.equal(mockApp);
+  
+    describe('bucket(invalid)', () => {
+      const expectedError = 'Bucket name not specified or invalid. Specify a valid bucket name via ' +
+          'the storageBucket option when initializing the app, or specify the bucket name ' +
+          'explicitly when calling the getBucket() method.';
+      const invalidNames = [null, NaN, 0, 1, true, false, '', [], [1, 'a'], {}, { a: 1 }, _.noop];
+      invalidNames.forEach((invalidName) => {
+        it(`should throw given invalid bucket name: ${ JSON.stringify(invalidName) }`, () => {
+          expect(() => {
+            const bucketAny: any = storage.bucket;
+            bucketAny(invalidName);
+          }).to.throw(expectedError);
+        });
+      });
     });
-
-    it('is read-only', () => {
-      expect(() => {
-        (storage as any).app = mockApp;
-      }).to.throw('Cannot set property app of #<Storage> which has only a getter');
+  
+    describe('bucket()', () => {
+      it('should return a bucket object', () => {
+        expect(storage.bucket().name).to.equal('bucketName.appspot.com');
+      });
     });
-  });
-
-  describe('bucket(invalid)', () => {
-    const expectedError = 'Bucket name not specified or invalid. Specify a valid bucket name via ' +
-        'the storageBucket option when initializing the app, or specify the bucket name ' +
-        'explicitly when calling the getBucket() method.';
-    const invalidNames = [null, NaN, 0, 1, true, false, '', [], [1, 'a'], {}, { a: 1 }, _.noop];
-    invalidNames.forEach((invalidName) => {
-      it(`should throw given invalid bucket name: ${ JSON.stringify(invalidName) }`, () => {
-        expect(() => {
-          const bucketAny: any = storage.bucket;
-          bucketAny(invalidName);
-        }).to.throw(expectedError);
+  
+    describe('bucket(valid)', () => {
+      it('should return a bucket object', () => {
+        expect(storage.bucket('foo').name).to.equal('foo');
       });
     });
   });
 
-  describe('bucket()', () => {
-    it('should return a bucket object', () => {
-      expect(storage.bucket().name).to.equal('bucketName.appspot.com');
-    });
-  });
+  describe('Emulator Support', () => {
```

Comment:
This is the only bit I expected. Why did we change other tests in this file?

Reply:
I split the emulator vs non-emulator tests into 2 different `describe` blocks. Git diff is showing the indentations as additions/deletions. These changes are no-op.

Reply:
Done.